Docs (utils): Add information about using callback with a single property. (#8274)
ckeditor/ckeditor5: packages/ckeditor5-utils/docs/framework/guides/deep-dive/observables.md

The property has been "renamed" in the binding and from now on, whenever `command.value` changes, the value of `button.isOn` will reflect it.

#### Processing a property value

> Another use case is processing a bounded value. For example: you may need to disable a button only if certain conditions are met. Passing a callback as the third parameter allows implementing custom logic.

Reviewer: ```suggestion
Another use case is processing the bound property value, for instance, when a button should be disabled only if certain conditions are met. Passing a callback as the third parameter allows implementing a custom logic.
```

> In the example below, the `isEnabled` property will be set `true` only when `command.value` equals `'heading1`.

Reviewer 2: ```suggestion
In the example below, the `isEnabled` property will be set to `true` only when `command.value` equals `'heading1`.
```

> In the example below, a button's `isEnabled` property will be set to `true` only when `command.value` will be set to `'heading1`.

Reviewer: ```suggestion
In the example below, the `isEnabled` property will be set `true` only when `command.value` equals `'heading1`.
```

```js
const command = editor.commands.get( 'heading' );

> button.bind( 'isOn', 'isEnabled' ).to( command, 'value', value => value === 'heading1' );

Reviewer: My gut tells me this won't work. I couldn't find this kind of binding in [tests](https://github.com/ckeditor/ckeditor5/blob/master/packages/ckeditor5-utils/tests/observablemixin.js#L260-L804).

Author: It's the "should work with callback #6" test  https://github.com/ckeditor/ckeditor5/blob/master/packages/ckeditor5-utils/tests/observablemixin.js#L645-L659 :wink: 

The example was inspired by this:

https://github.com/ckeditor/ckeditor5/blob/b17ae267083c41aefc80863a2e6cf16ffa5989cb/packages/ckeditor5-heading/src/headingui.js#L61

ps.: We should fix those test names as we find them. That's another technical dept.

Reviewer 3: I wish I knew better what are you guys talking about...

Author: @oleq wasn't sure if my additions actually work the way I described them. Which is a valid point since it wasn't documented before. I know it is possible as I already checked it in the tests (the test name could be improved) and the source code that uses this.

The problem was here that it wasn't documented explicitly, rather implicitly, and wasn't discoverable enough. Hence, the proposed additions to the docs.

Reviewer: The `.to()` part is OK. The `.bind( 'isOn', 'isEnabled' )` is wrong IMO as there are 2 property names here ( `isOn` will not work here). I think this will throw an error (but this needs to be confirmed).

Author: OH, yes it is a typo.

Author: I totally blanked on the first part.

Author: Fixed. Thanks.

> button.bind( 'isOn', 'isEnabled' ).to( command, 'value', value => value === 'heading1' );

Author: ```suggestion
button.bind( 'isOn' ).to( command, 'value', value => value === 'heading1' );
```
